dot/parachain/overseer/types.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package overseer
+
+import (
+	"sync"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+type Context struct {
+	Sender   Sender
+	Receiver chan any
+	wg       *sync.WaitGroup
+	stopCh   chan struct{}
+}
+
+type Sender interface {
+	SendMessage(msg any) error
+}
+
+type ActivatedLeaf struct {
+	Hash   common.Hash
+	Number uint32
+}
+
+type ActiveLeavesUpdate struct {
+	Activated *ActivatedLeaf
+}
```

Comment:
Can you provide comments for exported types? Is this the same terminology they use in polkadot?

Reply:
Yes, added comments, yes this is the terminology used in polkadot.